cli/src/cmd/test.rs

```diff
@@ -316,134 +318,100 @@ fn test<A: ArtifactOutput + 'static>(
     filter: Filter,
     json: bool,
     allow_failure: bool,
-    gas_reports: (bool, Vec<String>),
+    (gas_reporting, gas_reports): (bool, Vec<String>),
 ) -> eyre::Result<TestOutcome> {
     let verbosity = evm_opts.verbosity;
-    let gas_reporting = gas_reports.0;
     if gas_reporting && evm_opts.verbosity < 3 {
-        // force evm to do tracing, but dont hit the verbosity print path
+        // Enable tracing without hitting the verbosity print path
         evm_opts.verbosity = 3;
     }
     let mut runner = builder.build(project, evm_opts)?;
 
     if json {
         let results = runner.test(&filter, None)?;
-        let res = serde_json::to_string(&results)?; // TODO: Make this work normally
-        println!("{}", res);
+        // TODO: Make this work normally
+        println!("{}", serde_json::to_string(&results)?);
         Ok(TestOutcome::new(results, allow_failure))
     } else {
-        // TODO: Re-enable when ported
-        //let mut gas_report = GasReport::new(gas_reports.1);
+        let local_identifier = LocalTraceIdentifier::new(&runner.known_contracts);
```

Comment:
So we'd add a new identifier here for Etherscan, and then for each trace call `identify`? I wonder if there's a way to pull this "out" so that we can do 1 identification step for etherscan across _all_ traces

Reply:
We should be able to do something like:
```
if let Some(fork_url) = fork_url {
    let (mut all_addresses, mut _maybe_codes) = arena.addresses_iter().collect();
    let unidentified_contracts = all_addresses.iter().filter(|address| !identified_contracts.contains(address)).collect();
    let etherscan_identified = unidentified_contracts.par_iter().map(|contract| some_etherscan_contract_getter(contract)).collect();
    etherscan_identified.iter().for_each(|(addr, name, abi)| identified_contracts.insert(addr, (name, abi)));
 }
 ```

Reply:
note for debugger it gets a little more challenging because we have to not just get abi but get the contract code and rerun compilation pipeline and write the contracts to files :/

Reply:
Ah, right. We can get the source code, byte code and ABI just fine, but there is no way for us to get the sourcemap from the Etherscan API :(

Reply:
If we get the source code, we could always compile it with the right version and get it ourselves?